pkg/types/provider_ccip.go

```diff
@@ -44,3 +46,9 @@ type CCIPFactoryGenerator interface {
 	CCIPCommitFactoryGenerator
 	CCIPExecutionFactoryGenerator
 }
+
+type CCIPProvider interface {
+	services.Service
+	ChainAccessor() (ccipocr3.ChainAccessor, error)
+	ContractTransmitter() (ocr3types.ContractTransmitter[[]byte], error)
```

Comment:
Do these need to return errors? Or could the construction of the provider return error instead? Typically *Providers are superficial interfaces that just return internal things which were already constructed.